src/main/kotlin/graphics/scenery/utils/TGAReader.kt

```diff
@@ -0,0 +1,630 @@
+                                var color = -0x1
+                                if(colormapIndex >= 0) {
+                                    val index = 3 * colormapIndex + 18
+                                    val b = palette[index + 0].toInt() and 0xFF
+                                    val g = palette[index + 1].toInt() and 0xFF
+                                    val r = palette[index + 2].toInt() and 0xFF
+                                    val a = 0xFF
+                                    color = r shl rs or (g shl gs) or (b shl bs) or (a shl `as`)
+                                }
+                                pixels[width * (height - i - 1) + (width - j - 1)] = color
+                                j++
+                            }
+                            i++
+                        }
+                    }
+                } else {
+                    if(descriptor and UPPER_ORIGIN != 0) {
+                        // UpperLeft
+                        var i = 0
+                        while(i < height) {
+                            var j = 0
+                            while(j < width) {
+                                val colormapIndex = bytes[offset + width * i + j].toInt() and 0xFF - colormapOrigin
+                                var color = -0x1
+                                if(colormapIndex >= 0) {
+                                    val index = 3 * colormapIndex + 18
+                                    val b = palette[index + 0].toInt() and 0xFF
+                                    val g = palette[index + 1].toInt() and 0xFF
+                                    val r = palette[index + 2].toInt() and 0xFF
+                                    val a = 0xFF
+                                    color = r shl rs or (g shl gs) or (b shl bs) or (a shl `as`)
+                                }
+                                pixels[width * i + j] = color
+                                j++
+                            }
+                            i++
+                        }
+                    } else {
+                        // LowerLeft
+                        var i = 0
+                        while(i < height) {
+                            var j = 0
+                            while(j < width) {
+                                val colormapIndex = bytes[offset + width * i + j].toInt() and 0xFF - colormapOrigin
+                                var color = -0x1
+                                if(colormapIndex >= 0) {
+                                    val index = 3 * colormapIndex + 18
+                                    val b = palette[index + 0].toInt() and 0xFF
+                                    val g = palette[index + 1].toInt() and 0xFF
+                                    val r = palette[index + 2].toInt() and 0xFF
+                                    val a = 0xFF
+                                    color = r shl rs or (g shl gs) or (b shl bs) or (a shl `as`)
```

Comment:
Ooh, `shl`. Whyyyyyy? Why not go all the way with it:
```kotlin
color eq r shl rs or lpn g shl gs rpn or lpn b shl bs rpn or lpn a shl bt as bt rpn
```
Get rid of all those pesky symbols! So much better. :triumph:

Reply:
Haha, I found [KT-1440](https://youtrack.jetbrains.com/issue/KT-1440) where people argued ad nauseum about this, and @elect86 was among those commenting in favor of the symbols. :+1: Glad to know I'm not the only one who prefers `<<` over `shl`.

Reply:
That's the typical absurd example of denying perfectly normal features for the fear of abuse